pkg/internal/server/grpc/snaphotmetadata_handlers.go

```diff
@@ -36,16 +43,52 @@ func (s *Server) GetMetadataAllocated(req *api.GetMetadataAllocatedRequest, stre
 		return err
 	}
 
-	// TODO: Call CSI driver endpoint for changed block metadata
+	csiClient := csi.NewSnapshotMetadataClient(s.csiConnection())
+	csiReq := newCSIGetMetadataAllocatedRequest(ctx, snapshotHandle, req)
+
+	// Invoke CSI Driver's GetMetadataDelta gRPC and stream the response back to client
+	csiStream, err := csiClient.GetMetadataAllocated(ctx, csiReq)
+	if err != nil {
+		return err
+	}
+
+	done := make(chan bool)
+	errCh := make(chan error, 1)
+	go func() {
```

Comment:
Why is a goroutine necessary?  There does not appear to be any reason why the code cannot be inlined.

Reply:
good point, we can make it inlined, will fix this